cranelift-codegen/meta/src/cdsl/ast.rs

```diff
@@ -384,8 +384,51 @@ impl VarPool {
     pub fn get_mut(&mut self, index: VarIndex) -> &mut Var {
         self.pool.get_mut(index).unwrap()
     }
-    pub fn create(&mut self, name: &'static str) -> VarIndex {
-        self.pool.push(Var::new(name))
+    pub fn create(&mut self, name: impl Into<String>) -> VarIndex {
+        self.pool.push(Var::new(name.into()))
+    }
+}
+
+/// Contains constants created in the AST that must be inserted into the true [ConstantPool]
+/// (cranelift_codegen::ir::ConstantPool) when the legalizer code is generated. The constant data
+/// is named in the order it is inserted; inserting data using [insert]
+/// (cranelift_codegen_meta::cdsl::ast::insert) will avoid duplicates.
+pub(crate) struct ConstPool {
+    pool: Vec<Vec<u8>>,
+}
+
+impl ConstPool {
+    /// Create an empty constant pool.
+    pub fn new() -> Self {
+        Self { pool: vec![] }
+    }
+
+    /// Create a name for a constant from its position in the pool.
+    fn stringify(position: usize) -> String {
```

Comment:
Forgot to comment here: could we rename this function `constant_name_for`, maybe? Stringify may be confusing since it has other meanings (the macro just transforms any token tree into a Rust string, JS has JSON.stringify which does something different, etc.).